File: Rogers/src/fitme/schemas/health.py
"""Health Schemas"""
from pydantic import BaseModel
from typing import Optional, List
from datetime import date, datetime


class HealthMetrics(BaseModel):
    """健康指标"""
    weight: float
    height: float
    bodyFat: float
    bmi: float
    weightGoal: Optional[float] = None
    bmiStatus: str


class HealthMetricsResponse(BaseModel):
    """健康指标响应"""
    code: int = 200
    data: HealthMetrics


class CreateHealthMetricRequest(BaseModel):
    """创建健康指标请求"""
    weight: Optional[float] = None
    height: Optional[float] = None
    bodyFat: Optional[float] = None
    measureDate: date


class CreateHealthMetricResponse(BaseModel):
    """创建健康指标响应"""
    code: int = 200
    message: str = "记录成功"
    data: dict


class HealthMeasurement(BaseModel):
    """健康测量记录"""
    recordId: int
    weight: float
    bodyFat: float
    bmi: float
    measureDate: date
    createdAt: datetime


class HealthMeasurementsResponse(BaseModel):
    """历史测量记录响应"""
    code: int = 200
    data: List[HealthMeasurement]


class TrendPoint(BaseModel):
    """趋势数据点"""
    date: date
    value: float


class StatusSummary(BaseModel):
    """状态统计"""
    normal: int
    low: int
    high: int


class HealthReportSummary(BaseModel):
    """健康报表摘要"""
    avgWeight: float
    avgBmi: float
    weightChange: float
    statusSummary: StatusSummary


class HealthReport(BaseModel):
    """健康报表"""
    weightTrend: List[TrendPoint]
    bmiTrend: List[TrendPoint]
    summary: HealthReportSummary


class HealthReportResponse(BaseModel):
    """健康报表响应"""
    code: int = 200
    data: HealthReport

File: Rogers/web/src/services/health.ts
import api from '../utils/request'

export interface HealthMetrics {
  weight: number
  height: number
  bodyFat: number
  bmi: number
  weightGoal: number | null
  bmiStatus: string
}

export interface HealthMeasurement {
  recordId: number
  weight: number
  bodyFat: number
  bmi: number
  measureDate: string
  createdAt: string
}

export interface HealthReport {
  weightTrend: { date: string; value: number }[]
  bmiTrend: { date: string; value: number }[]
  summary: {
    avgWeight: number
    avgBmi: number
    weightChange: number
    statusSummary: { pass: number; low: number; high: number }
  }
}

export const healthApi = {
  getMetrics: (): Promise<HealthMetrics> =>
    api.get('/health/metrics'),

  createMetric: (data: { weight?: number; bodyFat?: number; measureDate: string }): Promise<{ recordId: number }> =>
    api.post('/health/metrics', data),

  getMeasurements: (limit: number = 10): Promise<HealthMeasurement[]> =>
    api.get('/health/measurements', { params: { limit } }),

  getReport: (period: string = 'week'): Promise<HealthReport> =>
    api.get('/health/report', { params: { period } }),

  exportData: (period: string = 'week', format: string = 'csv') =>
    api.get('/health/export', { params: { period, format }, responseType: 'blob' }),
}